library/libobject.mli

```diff
@@ -72,46 +72,54 @@ type substitutivity = Dispose | Substitute | Keep | Anticipate
 
 type object_name = Libnames.full_path * KerName.t
 
-type open_filter
+module Open_filter : sig
 
-type ('a,'b) object_declaration = {
-  object_name : string;
-  cache_function : 'b -> unit;
-  load_function : int -> 'b -> unit;
-  open_function : open_filter -> int -> 'b -> unit;
-  classify_function : 'a -> substitutivity;
-  subst_function :  substitution * 'a -> 'a;
-  discharge_function : 'a -> 'a option;
-  rebuild_function : 'a -> 'a;
-}
+  type t
+
+  val unfiltered : t
+
+  val make : finite:bool -> string CAst.t list -> t
+  (** Anomaly when the list is empty. *)
 
-val unfiltered : open_filter
+  module Category : sig
```

Comment:
Not sure this should be a submodule of Open_filter

Reply:
I agree. Should category go into `Data.t` ? I think we can do a bit better with this API in general, but I dunno.

Reply:
I would put it as Libobject.Category
No need to pile on inner modules

Reply:
Actually my motivation to put it into `Open_filter` is that actually `Open_filter` has access to the private implementation of `Category.t`. So I'm not sure what the best thing is here.

Reply:
> I would put it as Libobject.Category No need to pile on inner modules

I mean if we should store the category in the obj itself instead of encoding the filtering logic in the closure.

For the original motivation of putting it inside `Open_filter` see my comment about internals.

Reply:
We could imagine categories getting more complicated than 1 string, eg like in https://github.com/coq/coq/pull/14892#issuecomment-922903956
Not sure how much we should consider these dreams of the future while writing code today though